tools/jit/gen_jit_dispatch.py

```diff
@@ -222,46 +280,25 @@ def emit_decl_variant(decl, is_positional_arg, has_tensorlist):
                 pos_assignments.append(assign)
                 arguments.append(arg['name'])
             else:
+                attr_method = attr_of(jit_type_of(arg))
                 simple_type = arg['simple_type']
-
-                assert simple_type in ATTR_METHOD_MAP, (decl['name'], simple_type)
-                attr_method = ATTR_METHOD_MAP[simple_type]
-                assign = KW_ASSIGNMENT.substitute(type_cast=TYPE_CASTS.get(simple_type, simple_type),
+                assign = KW_ASSIGNMENT.substitute(type_cast=FROM_ATTRIBUTE.get(simple_type, simple_type),
                                                   name=arg['name'],
                                                   method=attr_method)
                 kw_assignments.append(assign)
-                attr_names.append('{}_{}'.format(arg['name'], attr_method))
                 arguments.append(arg['name'])
 
         call = get_invocation(decl, arguments, num_dynamic_inputs)
 
-        # Descriptor is a unique identifier for a particular overload of an op.
-        attr_names = sorted(attr_names)
-        num_inputs = '*' if has_tensorlist else static_inputs
-        descriptor = '-'.join([decl['name'], str(num_inputs)] + attr_names)
-
-        # If there are two overloads with the same descriptor, that differ only by a type of a
-        # single argument, where one of them takes a tensor, while another one takes an
-        # at::Scalar as a positional scalar arg, then prefer the tensor overload.
-        # It should get broadcasted correctly.
-        if descriptor in skip_scalar_overload:
-            if any(decl['arguments'][idx]['simple_type'] in {'Scalar', 'double'}
-                   for idx in skip_scalar_overload[descriptor]):
-                return
-
         returns = decl['returns']
         all_scalars = all(r['dynamic_type'] != 'TensorList' for r in returns)
-        num_outputs = str(len(returns)) if all_scalars else 'UNKNOWN_OUTPUTS'
 
-        constructor = CONSTRUCTOR.substitute(descriptor=descriptor, name=decl['name'],
-                                             call=call,
+        constructor = CONSTRUCTOR.substitute(name=decl['name'],
+                                             call=[call],
```

Comment:
why is this a singleton array?

Reply:
`substitute` handles newline insertion differently depending of whether it is passed a list (it will indent to current indent level), or a string (it won't do anything fancy). I added a comment.